### ISSUE TITLE
Smartlook doesn't group any user sessions

### ISSUE THREAD

**Issue by AndrewMorsillo** [smartlook/smartlook-mobile-issue-tracker#4]
**Platform**
Android and iOS,

**Technology**
 React Native 0.63.2

**Affected devices [optional]**
This appears to only affect Android but we're not sure as we don't have an iOS public beta running at the moment.

**Smartlook SDK version**
 smartlook-react-native-wrapper 0.32.0

**Summary**
Smartlook doesn't group any user sessions together. Every session shows up as a new user. The sessions that fail to group together have the correct user id and email set (they are identical across all the sessions for a user that fail to be grouped).

This obviously greatly decreases the usefulness of Smartlook and this bug comes at a time when we desperately need Smartlook to work correctly (early beta testing, the information from Smartlook is critical to our process and if it fails our ad buys are mostly wasted $).

Other packages in use:
```
        "@notifee/react-native": "^0.12.2",
        "@react-native-community/async-storage": "^1.12.0",
        "@react-native-community/google-signin": "^4.0.3",
        "@react-native-community/netinfo": "^5.9.5",
        "@react-native-firebase/analytics": "^7.6.1",
        "@react-native-firebase/app": "^8.4.1",
        "@react-native-firebase/auth": "^9.1.2",
        "@react-native-firebase/crashlytics": "^8.4.2",
        "@react-native-firebase/firestore": "^7.7.0",
        "@react-native-firebase/functions": "^7.4.2",
        "@react-native-firebase/messaging": "^7.8.4",
        "@react-native-firebase/perf": "^7.4.2",
        "clevertap-react-native": "^0.4.5",
        "date-fns": "^2.15.0",
        "fast-safe-stringify": "^2.0.7",
        "instabug-reactnative": "^9.1.7",
        "lodash": "^4.17.20",
        "metro-minify-terser": "^0.63.0",

        "mobx": "^5.15.4",
        "mobx-react": "^6.2.5",
        "mobx-state-tree": "^3.17.2",
        "mobx-utils": "^5.6.1",
        "moment": "^2.27.0",
        "observable-hooks": "^3.1.0",
        "react": "16.13.1",
        "react-native": "0.63.2",
        "react-native-animatable": "^1.3.3",
        "react-native-branch": "^4.4.0",
        "react-native-code-push": "^6.3.0",
        "react-native-device-log": "github:AndrewMorsillo/react-native-device-log#master",
        "react-native-exception-handler": "^2.10.8",
        "react-native-fbsdk": "^1.1.2",
        "react-native-popover-view": "^3.0.3",
        "react-native-restart": "^0.0.17",
        "react-native-safe-area-context": "^3.1.6",
        "react-native-sound": "^0.11.0",
        "react-native-svg": "^12.1.0",
        "react-native-vector-icons": "^7.0.0",
        "react-native-version-number": "^0.3.6",
        "react-native-xyz-scrollview": "github:AndrewMorsillo/react-native-directed-scrollview#master",
        "rxjs": "^6.6.2",
        "seedrandom": "^3.0.5",
        "smartlook-react-native-wrapper": "^0.32.0",
        "tinycolor2": "^1.4.1"
```

**Steps to reproduce [optional]**
Install smartlook on a react-native Android project and attempt to set user identifiers, for example
```
SmartlookAPI.setUserIdentifier(u.uid, {
                email: userEmail,
                name: userName,
                codepushVersion: versionNumber,
            })
```
Observe that subsequent sessions from a user are not grouped together on the Smartlook dashboard.

**Example Project [optional]**
N/A -- issue needs a project with a stream of new users to be apparent

**Relevant logs and/or screenshots**
Here's our smartlook dashboard filtered by one of my test accounts. I had several sessions on the same test device on the same day. Each one shows up as a new user with 1 session despite all of them having identical user id, email, device, and ip address.

![image](https://user-images.githubusercontent.com/858965/92049161-a686a100-ed57-11ea-8f77-c5bd1763b4f8.png)

**Comment by AndrewMorsillo** [comment]
I may have reinstalled the app between some, but not all, of those test runs. Regardless this issue is happening for all of our beta users who are certainly not reinstalling the app constantly. Shouldn't SL group the sessions anyway as long as they have the same user id?

I don't use `startNewSessionAndUser` or `resetSession` in our app code. Our integration is really straightforward, on app launch we do

```
SmartlookAPI = require("smartlook-react-native-wrapper")
SmartlookAPI.setupAndStartRecording("our-key-here")
SmartlookAPI.enableCrashlytics(true)
```

Then later once we have user auth data

```
 SmartlookAPI.setUserIdentifier(u.uid, {
                email: userEmail,
                name: userName,
                codepushVersion: versionNumber,
            })
```

Then on any analytics event `SmartlookAPI.trackCustomEvent(event, extra)` or `SmartlookAPI.trackNavigationEvent(screenName)` if it's a nav event. That's it.

**Comment by AndrewMorsillo** [comment]
@SenNeonoveNoci I can easily reproduce this using the example project from my other bug report https://github.com/AndrewMorsillo/TestSmartlookBug

Just run that project in release mode with a smartlook API key set, tap around the app, close it for a bit, open it again, then tap around some more. In a little while two separate sessions will appear in the Smartlook dashboard despite having the same user identifier. It appears this is an issue with the Smartlook backend. 

This is really negatively impacting us at a very important time. I hope we can get this resolved quickly. Why might two sessions not be attributed to the same user even though the user id, name, and email are identical? Is there anything else I can do to help?

**Comment by SenNeonoveNoci** [comment]
@AndrewMorsillo Thanks for the info and project. I'm gonna look at it and let you know ASAP.

**Comment by SenNeonoveNoci** [comment]
@AndrewMorsillo We have identified a cause of this problem. And working on a fix.

**Comment by SenNeonoveNoci** [comment]
@AndrewMorsillo We have identified the problem and fixed it.
We gonna release new version of the SDK and react native bridge at the start of next week. 
Sorry for the inconvenience.

**Comment by adamblack** [comment]
Hi @AndrewMorsillo ,

bug is fixed in new version of React native bridge, 0.33.0

**Comment by AndrewMorsillo** [comment]
@adamblack Thanks! Does this also potentially help with https://github.com/smartlook/smartlook-mobile-issue-tracker/issues/3 ?
